src/olympia/amo/management/commands/get_changed_files.py

```diff
@@ -0,0 +1,158 @@
+from django.core.management.base import BaseCommand
+
+import olympia.core.logger
+
+from olympia.addons.models import Addon, Preview
+from olympia.amo.utils import id_to_path
+from olympia.blocklist.utils import datetime_to_ts
+from olympia.files.models import File
+from olympia.hero.models import PrimaryHeroImage
+from olympia.users.models import UserProfile
+from olympia.versions.models import Version, VersionPreview
+
+
+def get_modified_folders(since, *, path, manager, datefield, id_to_path_breadth):
+    qs = manager.filter(**{f'{datefield}__gt': since}).values_list('id', flat=True)
+    return [
+        os.path.join(path, id_to_path(id_, breadth=id_to_path_breadth), '')
+        for id_ in qs
+    ]
+
+
+def collect_user_pics(since):
+    return get_modified_folders(
+        since,
+        path=os.path.join(settings.MEDIA_ROOT, 'userpics'),
+        manager=UserProfile.objects,
+        datefield='modified',
+        id_to_path_breadth=2,
+    )
+
+
+def collect_files(since):
+    path = settings.ADDONS_PATH
+    qs = File.objects.filter(modified__gt=since).values_list(
+        'version__addon_id', flat=True
+    )
+    return list({os.path.join(path, id_to_path(id_, breadth=2), '') for id_ in qs})
+
+
+def collect_sources(since):
+    return get_modified_folders(
+        since,
+        path=os.path.join(settings.MEDIA_ROOT, 'version_source'),
+        manager=Version.unfiltered,
+        datefield='modified',
+        id_to_path_breadth=1,
+    )
+
+
+def collect_xpi_uploads(since):
+    # all the files are in a single folder so return it
+    return [os.path.join(settings.ADDONS_PATH, 'temp')]
+
+
+def get_previews(since, path, manager):
+    out = set()
+    qs = manager.filter(created__gt=since).values_list('id', flat=True)
+    for preview_id in qs:
+        subdir = str(preview_id // 1000)
+        out = out | {
+            os.path.join(path, 'thumbs', subdir, ''),
+            os.path.join(path, 'full', subdir, ''),
+            os.path.join(path, 'original', subdir, ''),
+        }
+    return list(out)
+
+
+def collect_addon_previews(since):
+    return get_previews(
+        since, os.path.join(settings.MEDIA_ROOT, 'previews'), Preview.objects
+    )
+
+
+def collect_theme_previews(since):
+    return get_previews(
+        since,
+        os.path.join(settings.MEDIA_ROOT, 'version-previews'),
+        VersionPreview.objects,
+    )
+
+
+def collect_addon_icons(since):
+    path = os.path.join(settings.MEDIA_ROOT, 'addon_icons')
+    qs = Addon.unfiltered.filter(modified__gt=since).values_list('id', flat=True)
+    return list({os.path.join(path, str(preview_id // 1000), '') for preview_id in qs})
+
+
+def collect_editoral(since):
+    return (
+        [os.path.join(settings.MEDIA_ROOT, 'hero-featured-image')]
+        if PrimaryHeroImage.objects.filter(modified__gt=since).exists()
+        else []
+    )
+
+
+def collect_sitemaps(since):
+    return [settings.SITEMAP_STORAGE_PATH]  # sitemaps change each day
```

Comment:
We should probably skip them for now and just re-create them on the other side when doing the migration.